### PR TITLE
Use testcontainers for solr

### LINKED ISSUE #1472: Use testcontainers for solr tests
Part of #1014

Testcontainers doc: https://www.testcontainers.org/

Example PR: https://github.com/JanusGraph/janusgraph/pull/1409

Partial implementation: https://github.com/GDATASoftwareAG/janusgraph/tree/testcontainers-solr

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #1472
Part of #2260

FYI: kerberos tests are disabled as before on travis ci. I will create a ticket to activate kerberos tests again. #2311

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?

### CODE REVIEW

**Review by porunov — APPROVED** [review]
LGTM. Thank you @farodin91 !

**Comment by li-boxuan** [comment]
Not sure why but I am unable to run any Solr test locally (using Intellij). I am using MacOS Big Sur (not M1 chip). I can run CQL tests locally though.

The error is:
```
java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration

	at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$7(DockerClientProviderStrategy.java:215)
```

Tried different testcontainers version (the recent two versions) and it did not help.

**Comment by farodin91** [comment]
I'm unable to test these changes under an apple m1.

File: janusgraph-solr/src/test/java/org/janusgraph/diskstorage/solr/BerkeleySolrTest.java [hunk 1]
```diff
@@ -16,26 +16,21 @@
...
-/**
- * @author Matthias Broecheler (me@matthiasb.com)
- */
+import static org.janusgraph.BerkeleyStorageSetup.getBerkeleyJEConfiguration;
 
+@Testcontainers
 public class BerkeleySolrTest extends SolrJanusGraphIndexTest {
 
+    @Container
+    protected static JanusGraphSolrContainer solrContainer = new JanusGraphSolrContainer();
```
Comment (li-boxuan) [inline janusgraph-solr/src/test/java/org/janusgraph/diskstorage/solr/BerkeleySolrTest.java:28]: Now that we are using test containers for Solr tests, do you want to update TESTING.md in this PR?

**Comment by farodin91** [comment]
@li-boxuan I changed the testing.md and did a small cleanup.

**Review by li-boxuan — APPROVED** [review]
LGTM, thank you!

**Pull request merged**